Lib/datetime.py

```diff
@@ -407,18 +466,23 @@ def _check_time_fields(hour, minute, second, microsecond, fold):
         raise ValueError('second must be in 0..59', second)
     if not 0 <= microsecond <= 999999:
         raise ValueError('microsecond must be in 0..999999', microsecond)
+    if not 0 <= submicrosecond <= 999:
```

Comment:
`_index` is used to check all the other values are in fact int-like up above, but `submicrosecond` is not checked in such a fashion. Surely it should be?

Reply:
fixed